scripts/library.sh

```diff
@@ -332,10 +332,39 @@ function start_latest_knative_serving() {
   header "Starting Knative Serving"
   subheader "Installing Knative Serving"
   echo "Installing Serving from ${KNATIVE_SERVING_RELEASE}"
-  kubectl apply -f ${KNATIVE_SERVING_RELEASE} || return 1
+  # Some CRDs defined in serving YAML are also referenced by other components in serving. As it takes
+  # time for CRDs to become effective, there is a race condition between when the CRDs are effective
+  # and when the resources that references those CRDs are created.
+  # The current workaround is to re-apply serving.yaml if it fails. Remove the retry logic after the
+  # race condition is fixed.
+  execute_with_retries 30 3 kubectl apply -f ${KNATIVE_SERVING_RELEASE} || return 1
```

Comment:
This is a workaround. I think it would be simpler and more readable if you inline the 3 retries instead of writing a function for that. Also, can we make it work with a single retry?

Reply:
Updated it to use single retry. The single retry should be sufficient as long as there's enough time between the two install. Change the wait to sleep 1 minute for now and will increase it if it's still flaky.